documentcloud/documents/serializers.py

```diff
@@ -219,6 +224,20 @@ def __init__(self, *args, **kwargs):
         self._init_presigned_url(request, self.user, is_document, is_list)
         self._init_change_ownership(request, self.user, is_document, is_list)
 
+    @property
+    def _expandable_fields(self):
+        """Revisions are only expandable if you have edit access to the document"""
+        expandable_fields = super()._expandable_fields
+        request = self.context.get("request")
+        if (
+            not request
+            or not self.instance
+            or not request.user.has_perm("documents.change_document", self.instance)
+        ):
+            expandable_fields = expandable_fields.copy().pop("revisions")
```

Comment:
Do we want to use `copy.deepcopy` here? When I was looking at DRF flex fields a while back, I remember them sometimes being more than a level deep.

Reply:
This is a good question, but in this case the only change I am ever making is removing the "revisions" key, which is only a change at the top level.  Therefore, using a normal copy is sufficient.

Reply:
If later on, a change alters something other than revisions, that could cause an unexpected change to the original. This might be a reasonable risk, but sometimes this creates a class of bug that is hard to figure out.

Reply:
Since the performance cost should be negligible in this case, I will add the deepcopy to future proof the code